docs/devguide/testing.asciidoc

```diff
@@ -0,0 +1,67 @@
+[[testing]]
+=== Testing
+
+Beats has a various sets of tests. This guide should help to understand how the different test suites work, how they are used and new tests are added.
+
+In general there are two major test suites:
+
+* Tests written in Golang
+* Tests written in Python
+
+The tests written in Golang use the https://golang.org/pkg/testing/[Golang Testing package]. The tests written in Python depend on http://nose.readthedocs.io/en/latest/[nosetests] and required a compiled and executable binary from the Golang code. The python test run a beat with a specific config and params and either check if the output is as expected or if the correct things show up in the logs.
+
+For both of the above test suites so called integration tests exists. Integration tests in Beats are tests which require an external system like Elasticsearch to test if the integration with this service works as expected. Beats provides in its testsuite docker containers and docker-compose files to start these environments but a developer can run the required services also locally.
```

Comment:
Currently in beats there is no long running test which does not require an external system.  So in the current implementation the two things are coupled together. I can think of two options where we actually have long running tests but we cheated a bit by putting them into a different category. There are the benchmark tests which and there are the load tests we have. But both are not run when the integration tests are run.

It probably also comes down to what means "long running". 

In my head to simplify the above I would say we have 2 types of tests in Golang: 

* Tests which do not require an environment
* Tests which require an environment

All tests which do not require an environment are currently "short" running.